### DISCUSSION TITLE
The build of `picos`

### DISCUSSION THREAD

**Discussion by dinosaure** [ocaml-multicore/picos#34] [General]
Currently, `picos` trusts a lot on `dune` and use some mechanisms which are not probably available into `ocaml/ocaml` which can make integrating it into the standard library more difficult. It will be really nice to keep the simplest layout possible (like a module or some modules which can be compiled and usable only via `ocamlopt`) to make it easier to integrate into more restricted contexts (like `ocaml/ocaml` or into some projects that they don't use `dune`).

**Comment by polytypic** [comment]
Hmm...  Good point.  I believe that after the extra dependencies to `unix` and some other libraries are dropped ( #32 ) it should become easier to build the core library although we probably still want OCaml 4 and OCaml 5 specific implementation details (as there are no effects in OCaml 4).  Let's see how things look like at that point &mdash; I don't have the full design in my head yet.
